docs/src/pages/customization/unstyled-components/unstyled-components.md

```diff
@@ -0,0 +1,139 @@
+
+- `components` - an object that allows you to override subcomponents (slots) used by the unstyled "host" component. Each host component will at least have the `Root` slot. Many complex components have more slots. You can either provide a custom component or an HTML tag there.
+
+  ```jsx
+  <BadgeUnstyled components={{ Root: 'div', Badge: MyCustomBadge }} />
+  ```
+
+- `component` - a shortcut to `components.Root`. Note that if both `components.Root` and `component` are specified, `component` takes precedence.
+
+  ```jsx
+  <BadgeUnstyled component="div" />
+  ```
+
+- `componentsProps` - an object with each slot's props. Even if you don't override a specific slot (with `components` or `component`), you can provide additional props for the default components. For example, if you want to add a custom CSS class to one of the slots' components, specify it in the `componentsProps`.
+
+  ```jsx
+  <BadgeUnstyled componentsProps={{ input: { className: 'my-badge' } }} />
+  ```
+
+Additionally, all the extra props placed on the host component are propagated into the root component (just as if they were placed in `componentsProps.root`). These two examples are equivalent:
+
+```jsx
+<BadgeUnstyled id="badge1">
+```
+
+```jsx
+<BadgeUnstyled componentsProps={{ root: { id: 'badge1' } }}>
+```
+
+## Styling the unstyled
+
+There are several levels of customization available:
+
+### Applying custom CSS rules to unstyled components
+
+If you're happy with the structure of the HTML rendered by the unstyled component by default, you can just apply custom styles to the classes used by the component. Each component has its own set of classes. Some are static - present on different parts of the component all the time, and some are applied conditionally (like `Mui-disabled`, applied when a component is disabled).
+
+Each component's API documentation lists all classes that the component uses. Additionally, you can import a `[componentName]Classes` object that describes all the classes a given component uses (see the demo below for an example).
+
+{{"demo": "pages/customization/unstyled-components/styling-customCss.js"}}
+
+### Overriding the unstyled components' slots
+
+If you don't want to rely on default components, you can override them with your own. This makes it possible to provide a styled component. Each unstyled component has a specific set of "slots" - that is subcomponents that you can override.
+
+Let's take a [SwitchUnstyled](../api/switchUnstyled) as an example. It has three slots: `Root`, `Thumb` and `Input`. The demo below shows how to create a styled component (using [System](system/styled/) in this case, but it could well be any other solution)
+
+{{"demo": "pages/customization/unstyled-components/styling-slots.js"}}
+
+The components you pass in the `components` prop receive the `styleProps` prop from the top-level (host) component. By convention it contains all props passed in to the host component, merged with its internal state.
```

Comment:
- host component could be confused with host element https://reactjs.org/docs/implementation-notes.html#mounting-host-elements but the two notions seems distinct enough. react host = platform element, MUI host = owner of the slot.
- "internal" might feel private

```suggestion
The components you pass in the `components` prop receive the `styleProps` prop from the top-level component (host).
By convention, it contains all props passed to the host, merged with its rendering "state".
```